src/Symfony/Component/Finder/Iterator/PathFilterIterator.php

```diff
@@ -35,7 +35,7 @@ public function accept()
         // should at least not match one rule to exclude
         foreach ($this->noMatchRegexps as $regex) {
             if (preg_match($regex, $filename)) {
-                return false;
+                return $this->acceptChildren = false;
```

Comment:
this should not memoize: a folder can be excluded without excluding its children, depending of the regex you write (the path filtering is applied after flattening)